internal/tools/aws/cluster_management.go

```diff
@@ -235,13 +235,17 @@ func (a *Client) GetAndClaimVpcResources(cluster *model.Cluster, owner string, l
 	for _, vpc := range vpcs {
 		clusterResources, err := a.getClusterResourcesForVPC(*vpc.VpcId, *vpc.CidrBlock, logger)
 		if err != nil {
-			logger.Warn(err)
+			logger.WithError(err).WithField("vpc", *vpc.VpcId).Warn("Couldn't get cluster resources for VPC")
 			continue
 		}
 
 		err = a.claimVpc(clusterResources, cluster, owner, logger)
 		if err != nil {
-			return clusterResources, err
+			logger.WithError(err).WithField("vpc", *vpc.VpcId).Warn("Couldn't claim VPC")
+			if err := a.releaseVpc(cluster, logger); err != nil {
```

Comment:
This will not work because of the same reason. The `CloudClusterID` tag is missing.

Reply:
Yeah, we should test this out at least first in the dev account by manually messing with the tags. I am thinking there are a few scenarios where this will fail, but testing will validate that.